sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/QueryValidationTests.java

```diff
@@ -317,6 +325,132 @@ public void queryPlanCacheSinglePartitionParameterizedQueriesCorrectness() {
 
     }
 
+    @Test(groups = {"simple"}, timeOut = TIMEOUT * 10)
+    public void splitQueryContinuationToken() {
```

Comment:
don't we actually need a partition split to happen to simulate the partition split continuation token scenario?

Reply:
Yeah we need a split. So we are triggering a partition split in the test and waiting for it to happen and test moves forward only after asserting the same.
assertThat(partitionKeyRangesAfterSplit.size()).isGreaterThan(partitionKeyRanges.size())
             .as("Partition ranges should increase after split");

Reply:
partition split takes 15 minutes in prod to complete. no? 

does this mean the test takes 15 min to complete?

Reply:
If there are no writes, then I think it takes less than a minutes. 

---
In reply to: [561157054](https://github.com/Azure/azure-sdk-for-java/pull/18698#discussion_r561157054) [](ancestors = 561157054)

Reply:
When I run this test locally on prod endpoint, it takes about 8 - 10  10s sleep cycles which should mean that it completes in <100s